### PR TITLE
⌗112217 In Several Cases, Ints Should Be Floats

### PR DESCRIPTION
**Pull request by ggwicz**
**Ticket:** [**⌗112217**](http://central.tri.be/issues/112217)

**Before**|**After**
-----|-----
before|after

### CODE REVIEW

File: src/Tribe/Commerce/PayPal/Endpoints/Success_Template.php [hunk 1]
```diff
@@ -124,7 +124,8 @@ public function get_template_data( array $template_data = array() ) {
 				continue;
 			}
 
-			$ticket_price = (int) get_post_meta( $ticket_id, '_price', true );
+			$raw_ticket_price = get_post_meta( $ticket_id, '_price', true );
+			$ticket_price     = floatval( $raw_ticket_price );
```
Comment (lucatume) [inline src/Tribe/Commerce/PayPal/Endpoints/Success_Template.php]: why `floatval` and not `(float)`? The latter does the same thing and is faster.

File: src/Tribe/Commerce/PayPal/Order.php [hunk 2]
```diff
@@ -695,7 +695,7 @@ public function get_line_total() {
 		$statuses = apply_filters( 'tribe_tickets_tpp_order_line_total_statuses', $statuses, $this );
 
 		if ( in_array( $this->status, $statuses ) ) {
-			return ! empty( $this->meta['mc_gross'] ) ? (int) $this->meta['mc_gross'] : 0;
+			return ! empty( $this->meta['mc_gross'] ) ? floatval( $this->meta['mc_gross'] ) : 0;
```
Comment (lucatume) [inline src/Tribe/Commerce/PayPal/Order.php]: Same here: `(float)` in place of `floatval`.

File: src/Tribe/Commerce/PayPal/Orders/Sales.php [hunk 3]
```diff
@@ -53,10 +55,10 @@ public function get_revenue_for_ticket( Tribe__Tickets__Ticket_Object $ticket )
 	 *
 	 * @param Tribe__Tickets__Ticket_Object $ticket
 	 *
-	 * @return int
+	 * @return float
 	 */
 	protected function get_unfiltered_revenue_for_ticket( Tribe__Tickets__Ticket_Object $ticket ) {
-		return (int) $ticket->price * $ticket->qty_sold();
+		return floatval( $ticket->price * $ticket->qty_sold() );
```
Comment (lucatume) [inline src/Tribe/Commerce/PayPal/Orders/Sales.php]: `(float)` in place of `floatval`

File: src/Tribe/Commerce/PayPal/Orders/Sales.php [hunk 4]
```diff
@@ -303,10 +305,10 @@ public function get_orders_for_post( $post_id, array $ticket_ids = null ) {
 	 *
 	 * @param Tribe__Tickets__Ticket_Object $ticket
 	 *
-	 * @return int
+	 * @return float
 	 */
 	protected function get_ticket_completed_total( Tribe__Tickets__Ticket_Object $ticket ) {
-		return (int) $ticket->qty_sold() * (int) $ticket->price;
+		return floatval( $ticket->qty_sold() * $ticket->price );
```
Comment (lucatume) [inline src/Tribe/Commerce/PayPal/Orders/Sales.php]: `(float)` in place of `floatval`

File: src/Tribe/Commerce/PayPal/Orders/Sales.php [hunk 5]
```diff
@@ -329,10 +331,10 @@ protected function get_ticket_completed_qty( Tribe__Tickets__Ticket_Object $tick
 	 *
 	 * @param Tribe__Tickets__Ticket_Object $ticket
 	 *
-	 * @return int
+	 * @return float
 	 */
 	protected function get_ticket_not_completed_total( Tribe__Tickets__Ticket_Object $ticket ) {
-		return (int) $ticket->qty_pending() * (int) $ticket->price;
+		return floatval( $ticket->qty_pending() * $ticket->price );
```
Comment (lucatume) [inline src/Tribe/Commerce/PayPal/Orders/Sales.php]: `(float)` in place of `floatval`

**Comment by zbtirrell** [comment]
I love the screenshots in this pull request!

**Comment by ggwicz** [comment]
Feedback integrated; this is ready for another review! 🔍

**Comment by mitogh** [comment]
Some conflicts to be solved

**Comment by ggwicz** [comment]
Conflicts addressed, ready for another review 🔍

**Pull request merged**